packages/headless/src/controllers/search-status/headless-search-status.ts

```diff
@@ -0,0 +1,53 @@
+import {Engine} from '../../app/headless-engine';
+import {buildController, Controller} from '../controller/headless-controller';
+import {SearchSection} from '../../state/state-sections';
+
+/**
+ * The `SearchStatus` controller provides information on the status of the search.
+ */
+export interface SearchStatus extends Controller {
+  /**
+   * The state of the SearchStatus controller.
+   * */
+  state: SearchStatusState;
+}
+
+/**
+ * A scoped and simplified part of the headless state that is relevant to the `SearchStatus` controller.
+ */
+export interface SearchStatusState {
+  /**
+   * Determines if a search is in progress.
+   */
+  isLoading: boolean;
+  /**
+   * Determines if there are results available for the last executed query.
+   */
+  hasResults: boolean;
+  /**
+   * Determines if a first search has been executed.
+   */
+  firstSearchExecuted: boolean;
+}
+
+/**
+ * Creates a `SearchStatus` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @returns A `SearchStatus` controller instance.
+ * */
+export function buildSearchStatus(engine: Engine<SearchSection>): SearchStatus {
+  const controller = buildController(engine);
+
+  return {
+    ...controller,
+
+    get state() {
+      return {
+        isLoading: engine.state.search.isLoading,
+        hasResults: !!engine.state.search.results.length,
```

Comment:
Could instead do `hasNoResults` ? 🤔 

Typically you are more interested in the edge case that there's no results returned for a particular query, versus the opposite.

Reply:
I find using the positive avoids running into double-negatives. For example, if someone wants to do something when there _are_ results, they would have to write not-not:

```
if (!this.hasNoResults) {
    ...
}
```